config/v1/types_infrastructure.go

```diff
@@ -1417,6 +1419,89 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// When set, the failure domains defined here may be used to spread Machines across
+	// prism element clusters to improve fault tolerance of the cluster.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+}
+
+// NutanixFailureDomain configures failure domain information for the Nutanix platform.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	Name string `json:"name"`
+
+	// prismCentralEndpoint holds the endpoint (address, port) to access the Nutanix Prism Central.
+	// If this is not configured, it means this failure domain uses the default prism_central
+	// endpoint as configured in the nutanix platformSpec.
+	// +optional
+	PrismCentralEndpoint *NutanixPrismEndpoint `json:"prismCentralEndpoint,omitempty"`
+
+	// prismElementEndpoint holds the endpoint address and port to access the Nutanix Prism Element.
+	// When a cluster-wide proxy is installed, by default, this endpoint will be accessed via the proxy.
+	// Should you wish for communication with this endpoint not to be proxied, please add the endpoint to the
+	// proxy spec.noProxy list.
+	// +optional
+	PrismElementEndpoint *NutanixPrismElementEndpoint `json:"prismElementEndpoint,omitempty"`
+
+	// cluster is to identify the cluster (the Prism Element under management of the Prism Central),
+	// in which the Machine's VM will be created. The cluster identifier (uuid or name) can be obtained
+	// from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnets holds a list of identifiers (one or more) of the cluster's network subnets
+	// for the Machine's VM to connect to. The subnet identifiers (uuid or name) can be
+	// obtained from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
```

Comment:
Can we also put an upper limit on this? How many subnets are supported/do we expect a user to reasonably use?

Reply:
Ok, so, we only support one item, you can add a maximum list length of 1 here then, and in the future, if we need to allow more, can increase that limit.

So it should be min and max 1 for now from what you've said

Reply:
Added "validation:MaxItems=1" for now.